src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -13,12 +14,125 @@ namespace Aspire.Hosting.ApplicationModel;
 /// </summary>
 public class ResourceNotificationService(ILogger<ResourceNotificationService> logger)
 {
+    private static readonly TimeSpan s_defaultWaitForResourceStateTimeout = TimeSpan.FromSeconds(60);
```

Comment:
Lets not have a default timeout.

Reply:
Really? Hmmm. I feal uneasy about having methods that just hang forever so easily.

Reply:
See all of the methods on this class, pass a cancellation token if you want a timeout 😄

Reply:
The other methods on this class are async enumerable though so they're consumption pattern is a bit different. But if you feel strongly about it I'm happy to remove it and put the timeout in the test templates instead.